classes/class-p4-search.php

```diff
@@ -244,12 +304,115 @@ protected function check_cache( $cache_key, $cache_group ) {
 			// If cache key expired then retrieve results once again and re-cache them.
 			if ( false === $this->posts ) {
 				$this->posts = $this->get_timber_posts();
+
 				if ( $this->posts ) {
 					wp_cache_add( $cache_key, $this->posts, $cache_group, self::DEFAULT_CACHE_TTL );
 				}
 			}
 		}
 
+		/**
+		 * Fetches archived results from ArchiveIt.
+		 *
+		 * @param string $search_query search query entered by user.
+		 *
+		 * @return array of archived results.
+		 */
+		protected function get_archived_results( $search_query ) {
+
+			$group           = 'archive-search';
+			$subgroup        = 'all';
+			$cache_key_set   = $this->prepare_keys_for_cache( urldecode( $search_query . '-archived' ), $group, $subgroup );
+			$archive_results = wp_cache_get( $cache_key_set->key, $cache_key_set->group );
+
+			if ( ! $archive_results ) {
+
+				/**
+				 * Parameters
+				 * q = search query.
+				 * s = site.
+				 * h = hits per site (0 = all).
+				 * n = number of results. defaults to 10, so making it 100 should get enough.
+				 */
+				$archive_url = "https://archive-it.org/search-master/opensearch?q=$search_query&s=p3-raw.greenpeace.org&h=0&n=100";
+
+				$archive_response = wp_remote_get( $archive_url, array( 'timeout' => 100 ) );
+
+				$xml_archive_response_body = wp_remote_retrieve_body( $archive_response );
+
+				$response_body = new SimpleXMLElement( $xml_archive_response_body );
+
+				$filtered_parsed_results = $this->get_filtered_parsed_archived_results( $response_body );
+
+				if ( 0 == count( $filtered_parsed_results ) ) {
+					return [];
+				}
+
+				$archive_results = $filtered_parsed_results;
+
+				wp_cache_add( $cache_key_set->key, $filtered_parsed_results, $cache_key_set->group, self::DEFAULT_CACHE_TTL );
+			}
+			return $archive_results;
+		}
+
+		/**
+		 * Filter archived results by current domain.
+		 *
+		 * @param object $results of all results fetched from archive.
+		 *
+		 * @return array of archived results filtered by the current site.
+		 */
+		protected function get_filtered_parsed_archived_results( $results ) {
+			$domain = $this->get_domain();
+
+			$wayback_prefix = 'https://wayback.archive-it.org/' . self::ARCHIVE_COLLECTION . '/';
+
+			$filtered_parsed_items = [];
+
+			foreach ( $results->channel->item as $item ) {
+				if ( strpos( (string) $item->link, $domain ) !== false ) {
+					$parsed_item            = new stdClass;
```

Comment:
Need to use parenthesis when instantiating new object (i.e. `$parsed_item = new stdClass();`)